modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -81,31 +103,68 @@ def teardown(self):
     @cluster(num_nodes=NUM_NODES)
     @parametrize(version=str(DEV_BRANCH))
     @parametrize(version=str(LATEST_2_7))
-    def test_tcp(self, version):
+    def test_tcp_not_coordinator_single(self, version):
+        """
+        Test single-node-failure scenario (not the coordinator) with TcpDiscoverySpi.
+        """
+        return self.__simulate_nodes_failure(version, self.__properties(), 1)
```

Comment:
can we pass how many nodes to kill via @parametrize to reduce methods amount?

Reply:
> @matrix

We won't distinguish easily which test failed because, for example, 0 nodes to kill means to kill coordinator in internal routines. This reduces parameters count. Now we clearly  see it by test name. There is a lot of parametrization to me. We got only 4 test functions. They look not so many. Makes sense?

Reply:
I see code duplication and we have to reduce it if possible.
@matrix made for this.

0 to stop coordinator looks counter-intuitive to me, what about the case when we have kill coordinator and some node?
How to kill node after coordinator in one case and before at another?
matrix with a list of nodes to kill will be more clear? eg. [0, 3, 7], [5, 0, 2], [0, 5, 2].

Reply:
Yes, counter-intuitive. But you don't see it now. Now we have intuitive test names. It will become counter-inuitive with @matrix. This parametrization requires extra test organization. This PR do not include case for failing coordinator with some othe node. This could be done with some small changes and parameter. But looks like anohter ticket and PR to me.

Reply:
Ok for now.